app/TTSMenu.tsx

```diff
@@ -123,14 +130,14 @@ const TTSMenu = () => {
 
                         <Text style={{ ...styles.title, marginTop: 8 }}>Speaker</Text>
                         <Text style={styles.subtitle}>
-                            Speakers: {modelList.filter((item) => item.language === lang).length}
+                            Speakers: {languageList[lang]?.length > 0 ? filterModel(lang).length : filterModel(baseLang).length}
                         </Text>
 
                         <View style={{ marginTop: 8, marginBottom: 16, flexDirection: 'row' }}>
                             {modelList.length !== 0 && (
                                 <Dropdown
                                     value={currentSpeaker?.identifier ?? ''}
-                                    data={languageList[lang]}
+                                    data={languageList[lang]?.length > 0 ? languageList[lang] : languageList[baseLang]}
```

Comment:
Thanks for explaining the bug. However, it seems that the issue isn't exactly caused by matching the language exactly. The primary issue is due to the fact that <Dropdown> receives `undefined` as its data. Your fixes will work for `en` languages, but will likely fail for any other string, eg `zh` or `de`.

The cause of the issue `languageList[lang]` can possibly return `undefined`. 

Your fix above can still fail if both `languageList[lang]` and `languageList[baseLang]` are `undefined`.

The safe solution to this issue will be to update the TTS menu to have nullish coalescense when `languageList[lang]` does not exist.

I will push my own update for this issue, but thanks for pointing out the bug!